File: record-generator/processor/src/main/java/me/bristermitten/mittenlib/codegen/union/ResolvedUnionConstructor.java
package me.bristermitten.mittenlib.codegen.union;

import com.squareup.javapoet.ClassName;
import me.bristermitten.mittenlib.codegen.record.RecordConstructorSpec;
import me.bristermitten.mittenlib.codegen.record.RecordSpecLike;

public record ResolvedUnionConstructor(
        ClassName source,
        ClassName name,
        RecordConstructorSpec constructor
) implements RecordSpecLike {
}

File: record-generator/processor/src/main/scala/me/bristermitten/mittenlib/codegen/union/ResolvedUnionConstructor.scala
package me.bristermitten.mittenlib.codegen.union

import com.palantir.javapoet.ClassName
import me.bristermitten.mittenlib.codegen.record.RecordConstructorSpec
import me.bristermitten.mittenlib.codegen.record.RecordSpecLike

case class ResolvedUnionConstructor(source: ClassName, name: ClassName, constructor: RecordConstructorSpec)
  extends RecordSpecLike
